pylib/anki/_backend/genbackend.py

```diff
@@ -129,52 +110,58 @@ def get_input_assign(msg):
     return ", ".join(f"{f.name}={f.name}" for f in fields)
 
 
+# messages we don't want to unroll in codegen
+SKIP_UNROLL_INPUT = {
```

Comment:
Maybe we should discuss it now, because it's also related to the TODO you've commented on earlier.

How should we deal with situations, where `mediasrv` receives the message from JS, but still needs to do some modifications on the data?

1. Solution 1: I read there's a function `google.protobuf.json_format.MessageToDict`, which we could use to transform the message we receive from JS to a dict, modify it, and then pass it to the non-raw version.

2. Solution 2: We introduce a third variant, let's call it `_input`:

```python
def change_notetype_raw(self, input: bytes) -> bytes:
    return self._run_command(4, 15, input)

def change_notetype_input(self, *, input: anki.notetypes_pb2.ChangeNotetypeRequest) -> anki.collection_pb2.OpChanges:
    raw_bytes = self.change_notetype_raw(input.SerializeToString())
    output = anki.collection_pb2.OpChanges()
    output.ParseFromString(raw_bytes)
    return output

def change_notetype(self, *, note_ids: Sequence[int], new_fields: Sequence[int], new_templates: Sequence[int], old_notetype_id: int, new_notetype_id: int, current_schema: int, old_notetype_name: str) -> anki.collection_pb2.OpChanges:
    input = anki.notetypes_pb2.ChangeNotetypeRequest(note_ids=note_ids, new_fields=new_fields, new_templates=new_templates, old_notetype_id=old_notetype_id, new_notetype_id=new_notetype_id, current_schema=current_schema, old_notetype_name=old_notetype_name)
    return change_notetype_input(input)

```

Reply:
A third variant feels a bit much, and I don't think MessageToDict buys us anything here, since we can mutate the message directly - converting to a dict first would just be an extra step. I think we should just keep it simple:

- the simple, common case uses the non _raw variant
- for special cases, like when we need to modify the message, the caller is responsible for encoding/decoding the message themselves

So to fix the todo below, we could do:

```diff
diff --git a/qt/aqt/mediasrv.py b/qt/aqt/mediasrv.py
index 813624881..9d47b20c4 100644
--- a/qt/aqt/mediasrv.py
+++ b/qt/aqt/mediasrv.py
@@ -24,7 +24,7 @@ import aqt
 from anki import hooks
 from anki._vendor import stringcase
 from anki.collection import OpChanges
-from anki.decks import UpdateDeckConfigs
+from anki.decks import UpdateDeckConfigs, DeckConfigsForUpdate
 from anki.scheduler.v3 import NextStates
 from anki.utils import dev_mode
 from aqt.changenotetype import ChangeNotetypeDialog
@@ -382,8 +382,10 @@ def congrats_info() -> bytes:
 
 
 def get_deck_configs_for_update() -> bytes:
-    # TODO msg.have_addons = aqt.mw.addonManager.dirty
-    return aqt.mw.col._backend.get_deck_configs_for_update_raw(request.data)
+    config_bytes = aqt.mw.col._backend.get_deck_configs_for_update_raw(request.data)
+    configs = DeckConfigsForUpdate.FromString(config_bytes)
+    configs.have_addons = aqt.mw.addonManager.dirty
+    return configs.SerializeToString()
 
 
 def update_deck_configs() -> bytes:
```

It's not much more work than what we had, and makes it explicit that we're doing an extra decode/encode step.